src/Storages/StorageBuffer.cpp

```diff
@@ -1197,6 +1197,28 @@ void StorageBuffer::alter(const AlterCommands & params, ContextPtr local_context
     setInMemoryMetadata(new_metadata);
 }
 
+UInt64 checkUnderflowAndGetUInt64(const ASTPtr & arg, const String & arg_name)
+{
+    /**
+      * Do not force UInt64 type for args, it'll backward incampatible,
+      * there are exponential notation usages for shortness.
+      */
+    const auto & value = arg->as<ASTLiteral &>().value;
+    if (value.getType() > Field::Types::UInt64)
+    {
+        Int64 val = applyVisitor(FieldVisitorConvertToNumber<Int64>(), value);
```

Comment:
`Int64` has smaller positive range than `UInt64` so it seems it would be incorrect to convert it to `Int64` and then return it as a result. As this was done to check for negative value, then we could just manually check the value of `value.getType()` and for return value do visitor for `UInt64` as before.